test/ParserTest.cpp

```diff
@@ -5,43 +5,674 @@
-  ParserRAII parser(code);
-  Instruction const* pInstructions;
-  auto count = Parser_getInstructionSet(parser, &pInstructions);
-  vector<Instruction> instructions{pInstructions, pInstructions + count};
-  return std::invoke(std::forward<C>(cb), instructions);
-}
 } // namespace
 
 TEST(ParserTest, InstructionCountShouldBeValid) {
-  auto count = withInstructionsOf(R"(
+  auto code = R"(
+mov r0 r1;
+mov r1 r2;
+// abcd
+add r3 r4;
+add r3 2;
+// hah
+// hah2
+sub r2 0;
+mul r4, 3;
+)";
+
+  ParserRAII parser{code};
+  MockCpuRegisterMap<> regMap{};
+
+  ASSERT_EQ(6, parser.instructions(regMap.map()).size());
+}
+
+TEST(ParserTest, InstructionTypesShouldBeValid) {
+  auto code = R"(
+mov r0 r1;
+mov r1 r2;
+// abcd
+add r3 r4;
+add r3 2;
+// hah
+// hah2
+sub r2 0;
+mul r4, 3;
+)";
+
+  ParserRAII parser{code};
+  MockCpuRegisterMap<> regMap{};
+
+  ASSERT_EQ(instructions(
+      mov(), mov(), add(), add(), sub(), mul()
+  ), parser.instructions(regMap.map()));
+}
+
+TEST(ParserTest, InstructionConstantsShouldBeValid) {
+  auto code = R"(
+mov r0 r1;
+mov r1 54;
+// abcd
+add r3 4;
+add r3 2;
+// hah
+// hah2
+sub r2 0;
+mul r4, 3;
+)";
+
+  ParserRAII parser{code};
+  MockCpuRegisterMap<> regMap{};
+
+  ASSERT_EQ(instructions(
+      mov(), mov(any(), 54), add(any(), 4), add(any(), 2), sub(any(), 0), mul(any(), 3)
+  ), parser.instructions(regMap.map()));
+}
+
+TEST(ParserTest, InstructionConstantInBase2ShouldBeValid) {
+  auto code = R"(
+mov r0 0b1011;
+)";
+
+  ParserRAII parser{code};
+  MockCpuRegisterMap<> regMap{};
+
+  ASSERT_EQ(instructions(
+      mov(any(), 0b1011)
+  ), parser.instructions(regMap.map()));
+}
+
+TEST(ParserTest, InstructionConstantInBase8ShouldBeValid) {
+  auto code = R"(
+mov r0 0766;
+)";
+
+  ParserRAII parser{code};
+  MockCpuRegisterMap<> regMap{};
+
+  ASSERT_EQ(instructions(
+      mov(any(), 0766)
+  ), parser.instructions(regMap.map()));
+}
+
+TEST(ParserTest, InstructionConstantInBase16ShouldBeValid) {
+  auto code = R"(
+mov r0 0xDEAD;
+)";
+
+  ParserRAII parser{code};
+  MockCpuRegisterMap<> regMap{};
+
+  ASSERT_EQ(instructions(
+      mov(any(), 0xDEAD)
+  ), parser.instructions(regMap.map()));
+}
+
+TEST(ParserTest, InstructionConstantInvalidShouldYieldError) {
+  auto code = R"(
+mov r0 10abc;
+)";
+
+  try {
+    ParserRAII parser{code};
+  } catch (ParserException const& exception) {
+    ASSERT_EQ("10abc", exception.token());
+    ASSERT_EQ(2, exception.line());
+    ASSERT_EQ(8, exception.column());
+  }
+}
+
+TEST(ParserTest, CommasDoNothing) {
+  auto code = R"(
+mov r0 , 10;
+)";
+
+  ParserRAII parser{code};
+  MockCpuRegisterMap<> regMap{};
+  ASSERT_EQ(instructions(mov(any(), 10)), parser.instructions(regMap.map()));
+}
+
+TEST(ParserTest, CommasDoNothing2) {
+  auto code = R"(
+mov r0, 10;
+)";
+
+  ParserRAII parser{code};
+  MockCpuRegisterMap<> regMap{};
+  ASSERT_EQ(instructions(mov(any(), 10)), parser.instructions(regMap.map()));
+}
+
```

Comment:
Missed test cases:
```suggestion
TEST(ParserTest, CommasDoNothing3) {
  auto code = R"(
mov r0 ,10;
)";

  ParserRAII parser{code};
  MockCpuRegisterMap<> regMap{};
  ASSERT_EQ(instructions(mov(any(), 10)), parser.instructions(regMap.map()));
}

TEST(ParserTest, CommasDoNothing4) {
  auto code = R"(
mov r0,10;
)";

  ParserRAII parser{code};
  MockCpuRegisterMap<> regMap{};
  ASSERT_EQ(instructions(mov(any(), 10)), parser.instructions(regMap.map()));
}
```

Reply:
Added and addressed these two, not failing in branch head